flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1612,10 +1689,10 @@ def assert_calculated_changes(
 
     :param discovered_datasets: Collection of ``DiscoveredDataset`` to
         expose as local state.
+    :param api: ``IBlockDeviceAPI`` provider to use; by default a
```

Comment:
Strongly request that this not be made a parameter.

I think it's vital that calculate_changes is a pure function of it's arguments. We arguably could have done a different design to force that, but in lieu of a complete re-design this test infrastructure is the best thing we have to ensure we aren't making API calls during calculate_changes.

I don't think you actually use the parameter so it shouldn't be an issue?